core/unit_test/tools/include/ToolTestingUtilities.hpp

```diff
@@ -311,6 +311,8 @@ struct EventBase {
 template <class Derived>
 struct UniquelyIdentifiableEventType : public EventBase {
   uintptr_t kind() const override { return event_type_uid<Derived>(); }
+  // NOLINTNEXTLINE(bugprone-crtp-constructor-accessibility)
+  UniquelyIdentifiableEventType() = default;
```

Comment:
We cannot make `Derived` friend for this class ?

Reply:
This class isn't used in the same way as other CRTP classes. We have, e.g., 
```
template <typename Derived>
struct BeginOperation : public UniquelyIdentifiableEventType<Derived>
```
instead of
```
template <typename Derived>
struct BeginOperation : public UniquelyIdentifiableEventType<BeginOperation<Derived>>
```
so we don't know the class that would need to become a `friend`.

Reply:
Indeed I agree